test/flood_filler_test.exs

```diff
@@ -0,0 +1,112 @@
+defmodule FloodFillerTest do
+  use ExUnit.Case
+
+  test "all tiles with 0 adjacent bombs that are contiguous to the initial tile are revealed" do
+    board =
+      Board.new(3, 3).board
+      |> Board.reveal_tile({0, 0})
+
+    flood_filled_board = FloodFiller.flood_fill(board, {0, 0})
+
+    assert format(flood_filled_board) == [
+             ["0", "0", "0"],
+             ["0", "0", "0"],
+             ["0", "0", "0"]
+           ]
+  end
+
+  test "all contiguous tiles with 0 adjacent bombs are revealed, even if the initial tile has not been revealed yet" do
+    board = Board.new(3, 3).board
+
+    flood_filled_board = FloodFiller.flood_fill(board, {0, 0})
+
+    assert format(flood_filled_board) == [
+             ["0", "0", "0"],
+             ["0", "0", "0"],
+             ["0", "0", "0"]
+           ]
+  end
+
+  test "does not reveal contiguous tiles with 0 adjacent bombs if the initial tile has an adjacent bomb" do
+    board =
+      Board.new(3, 3).board
+      |> Board.replace_tile({0, 0}, Tile.new(:bomb))
+      |> AdjacentBombCount.set_adjacent_bomb_counts()
+      |> Board.reveal_tile({0, 1})
+
+    flood_filled_board = FloodFiller.flood_fill(board, {0, 1})
+
+    assert format(flood_filled_board) == [
+             ["B", "1", "?"],
+             ["?", "?", "?"],
+             ["?", "?", "?"]
+           ]
+  end
+
+  test "reveals the tiles with adjacent bombs that border the contiguous tiles with 0 adjacent bombs" do
+    board =
+      Board.new(3, 3).board
+      |> Board.replace_tile({0, 0}, Tile.new(:bomb))
+      |> AdjacentBombCount.set_adjacent_bomb_counts()
+      |> Board.reveal_tile({0, 2})
+
+    flood_filled_board = FloodFiller.flood_fill(board, {0, 2})
+
+    assert format(flood_filled_board) == [
+             ["B", "1", "0"],
+             ["1", "1", "0"],
+             ["0", "0", "0"]
+           ]
+  end
+
+  test "reveals contiguous tiles with 0 adjacent bombs on a small, non-square board" do
+    board =
+      Board.new(1, 4).board
+      |> Board.replace_tile({0, 0}, Tile.new(:bomb))
+      |> AdjacentBombCount.set_adjacent_bomb_counts()
+      |> Board.reveal_tile({0, 2})
+
+    flood_filled_board = FloodFiller.flood_fill(board, {0, 2})
+
+    assert format(flood_filled_board) == [
+             ["B", "1", "0", "0"]
+           ]
+  end
+
+  test "reveals contiguous tiles with 0 adjacent bombs for a very squiggly set of continuous squares" do
```

Comment:
lol, at least you improved the rest of the test names.

Reply:
@ecmendenhall this test name makes me laugh, which I think is a good enough reason to leave it in lol